NEWS.md

```diff
@@ -2,10 +2,20 @@
 
 ## New Features
 
+- The documentation was enhanced: (#2585)
+    - The default value of an argument is now displayed in the argument description.
+    - For complex functions each example has now a title, which is also shown in
+    the TOC, and a description. This enabled adding more examples without losing
+    readability. See `derive_extreme_records()` for an example.
+    - The output of the structured examples used for complex functions is
+    displayed in the help pages in RStudio.
+
```

Comment:
I think it might be a good idea to also then include under "Updates to documentation" a line for each of the functions that has undergone an examples revamp. So, in this PR, we'd add derive_extreme_records, but if we get to more functions by the release, we can list them there.

Reply:
Yes, item added.